Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2CameraSensorHook.cpp

```diff
@@ -0,0 +1,90 @@
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <RobotImporter/SDFormat/ROS2SensorHooks.h>
+
+#include <sdf/Camera.hh>
+#include <sdf/Sensor.hh>
+
+namespace ROS2::SDFormat
+{
+
+    SensorImporterHook ROS2SensorHooks::ROS2CameraSensor()
+    {
+        SensorImporterHook importerHook;
+        importerHook.m_sensorTypes =
+            AZStd::unordered_set<sdf::SensorType>{ sdf::SensorType::CAMERA, sdf::SensorType::DEPTH_CAMERA, sdf::SensorType::RGBD_CAMERA };
+        importerHook.m_supportedSensorParams =
+            AZStd::unordered_set<AZStd::string>{ ">update_rate",         ">camera>horizontal_fov", ">camera>image>width",
+                                                 ">camera>image>height", ">camera>clip>near",      ">camera>clip>far" };
+        importerHook.m_pluginNames = AZStd::unordered_set<AZStd::string>{ "libgazebo_ros_camera.so",
+                                                                          "libgazebo_ros_depth_camera.so",
+                                                                          "libgazebo_ros_openni_kinect.so" };
+        importerHook.m_supportedPluginParams = AZStd::unordered_set<AZStd::string>{};
+        importerHook.m_sdfSensorToComponentCallback = [](AZ::Entity& entity,
+                                                         const sdf::Sensor& sdfSensor) -> SensorImporterHook::ConvertSensorOutcome
+        {
+            auto* cameraSensor = sdfSensor.CameraSensor();
+            if (!cameraSensor)
+            {
+                return AZ::Failure(AZStd::string("Failed to read parsed SDFormat data of %s camera sensor", sdfSensor.Name().c_str()));
+            }
+
+            CameraSensorConfiguration cameraConfiguration;
+            cameraConfiguration.m_depthCamera = cameraSensor->HasDepthCamera();
+            cameraConfiguration.m_colorCamera = (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA) ? true : false;
+            cameraConfiguration.m_width = cameraSensor->ImageWidth();
+            cameraConfiguration.m_height = cameraSensor->ImageHeight();
+            cameraConfiguration.m_verticalFieldOfViewDeg =
+                cameraSensor->HorizontalFov().Degree() * (cameraConfiguration.m_height / cameraConfiguration.m_width);
+            if (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA)
+            {
+                cameraConfiguration.m_nearClipDistance = static_cast<float>(cameraSensor->NearClip());
+                cameraConfiguration.m_farClipDistance = static_cast<float>(cameraSensor->FarClip());
+            }
+            else
+            {
+                cameraConfiguration.m_nearClipDistance = static_cast<float>(cameraSensor->DepthNearClip());
+                cameraConfiguration.m_farClipDistance = static_cast<float>(cameraSensor->DepthFarClip());
+            }
+
+            SensorConfiguration sensorConfiguration;
+            sensorConfiguration.m_frequency = sdfSensor.UpdateRate();
+            if (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA)
```

Comment:
This is a bit confusing, are we creating color camera topics for something else other than sdf::SensorType::CAMERA, and depth topics for something else other than sdf::SensorType::DEPTH_CAMERA?

If not, it is clearer to replace != conditions with ==

Reply:
There is a third option: sdf::SensorType::RGBD_CAMERA, which consists of both color and depth sensors. Comment added.